models/login_source.go

```diff
@@ -291,22 +297,53 @@ func GetLoginSourceByID(id int64) (*LoginSource, error) {
 	return source, nil
 }
 
+// ResetNonDefaultLoginSources clean other default source flag
+func ResetNonDefaultLoginSources(source *LoginSource) error {
+	// update changes to DB
+	if _, err := x.NotIn("id", []int64{source.ID}).Cols("is_default").Update(&LoginSource{IsDefault: false}); err != nil {
+		return err
+	}
+	// write changes to local authentications
+	for i := range localLoginSources.sources {
+		localSource := &LoginSource{}
+		*localSource = *localLoginSources.sources[i]
+		if localSource.LocalFile != nil {
+			if localSource.ID != source.ID {
+				localSource.LocalFile.SetGeneral("is_default", "false")
+				if err := localSource.LocalFile.SetConfig(source.Cfg); err != nil {
+					return fmt.Errorf("LocalFile.SetConfig: %v", err)
+				} else if err = localSource.LocalFile.Save(); err != nil {
+					return fmt.Errorf("LocalFile.Save: %v", err)
+				}
+			}
+		}
+	}
+	// flush memory so that web page can show the same behaviors
+	localLoginSources.UpdateLoginSource(source)
+	return nil
+}
+
 // UpdateLoginSource updates information of login source to database or local file.
 func UpdateLoginSource(source *LoginSource) error {
 	if source.LocalFile == nil {
-		_, err := x.Id(source.ID).AllCols().Update(source)
-		return err
+		if _, err := x.Id(source.ID).AllCols().Update(source); err != nil {
+			return err
+		} else {
+			return ResetNonDefaultLoginSources(source)
+		}
+
 	}
 
 	source.LocalFile.SetGeneral("name", source.Name)
 	source.LocalFile.SetGeneral("is_activated", com.ToStr(source.IsActived))
+	source.LocalFile.SetGeneral("is_default", com.ToStr(source.IsDefault))
 	if err := source.LocalFile.SetConfig(source.Cfg); err != nil {
 		return fmt.Errorf("LocalFile.SetConfig: %v", err)
 	} else if err = source.LocalFile.Save(); err != nil {
 		return fmt.Errorf("LocalFile.Save: %v", err)
 	}
+	ResetNonDefaultLoginSources(source)
```

Comment:
Need to add error-checking here, or just `return ResetNonDefaultLoginSources(source)`